### PR TITLE
fix(kernel): honor the SwiGLU clamp limit in the Marlin MXFP4 MoE

### PR DESCRIPTION
**Pull request by FC-Li**
## Summary

`marlin_mxfp4_precomputed_moe_apply` ran its GEMM1 epilogue as a plain `silu_and_mul(intermediate1)` and never read `w.swiglu_arg`. Every other MoE backend (flashinfer, gluon, triton) applies the checkpoint's SwiGLU clamp from that attribute; on Hopper the Marlin path is the only MXFP4 MoE solution, so DeepSeek-V4.1-Flash (`swiglu_limit=10`) served all 40 routed-expert layers and the 3 DSpark draft MoEs with an unclamped activation. The reference implementation shipped with the checkpoint clamps `up` to `[-10, 10]` and `gate` from above before `silu(gate) * up`, and notes the clamps come from training. The shared expert already went through `SiluAndMul(10)` and was correct.

The same apply serves the non-EP, EP (`--enable-expert-parallel` with local experts masked and ranks summed) and DeepEP all-to-all layouts, so all three are fixed together.

## Changes

- `ops/moe/marlin/mxfp4.py`: new `_swiglu_limit(w)` reads the limit off `w.swiglu_arg` and passes it to `silu_and_mul`, whose Triton kernel already implements the asymmetric clamp. A SwiGLU `alpha` other than 1 or a non-zero `swiglu_beta` now raises instead of being silently dropped — Marlin has no epilogue for them, matching the Triton bf16 MoE's contract. Docstring documents the activation contract.
- `test/nvidia/ops/moe/test_marlin_mxfp4_apply.py`: a clamped-SwiGLU MXFP4 reference (per the checkpoint's `Expert.forward`) with weights scaled so gate/up routinely exceed the limit, parametrized over `ep_size` 1 and 2; the test asserts the clamp changes the answer by more than half the output magnitude so it discriminates. Plus unit cases for reading the limit and rejecting `alpha`/`beta`.

## Test plan

- [x] `pytest tokenspeed-kernel/test/nvidia/ops/moe/test_marlin_mxfp4_apply.py` on SM90: 13 passed (the existing SiTU / EP / SiLU cases are unaffected — they carry no `swiglu_arg`).
- [x] Reverting the one-line epilogue change makes both EP variants of the new test fail with 98.9% mismatched elements, so it catches the original bug.
- [x] `pre-commit run --all-files` clean.
- [ ] End-to-end A/B on a Hopper DeepSeek-V4.1-Flash EP8 deployment (GSM8K / long-context) after restart — expected neutral-to-better since this restores the trained activation.

Note for anyone running a model with a non-standard SwiGLU (`alpha != 1` or `swiglu_beta != 0`, e.g. gpt-oss) through Marlin on Hopper: that configuration was producing wrong activations silently and now fails at startup with a clear error.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/marlin/mxfp4.py [hunk 1]
```diff
@@ -64,6 +64,28 @@ def _block_size_m(num_tokens: int, top_k: int, num_experts: int) -> int:
...
+    """The checkpoint's SwiGLU clamp, or None for a plain ``silu(gate) * up``.
+
+    The clamp is the only SwiGLU knob this path can honor: ``silu_and_mul``
+    clips ``gate`` from above and ``up`` on both sides. A sigmoid multiplier
+    (``alpha``) or an up-branch offset (``swiglu_beta``) has no Marlin
+    epilogue, so it is rejected instead of silently dropped.
+    """
+    swiglu_arg = getattr(w, "swiglu_arg", None)
+    if swiglu_arg is None:
+        return None
```
[the remark this answers is not in the document]
Comment (FC-Li) [reply tokenspeed-kernel/python/tokenspeed_kernel/ops/moe/marlin/mxfp4.py]: Good catch — `swiglu_beta` is stored on the module independently of `swiglu_arg`. Fixed in 075ba401: alpha/beta are validated before the missing-`swiglu_arg` return, and the rejection test gained the `(swiglu_arg=None, swiglu_beta=1.0)` case.
